incubator/hnc/pkg/stats/metrics.go

```diff
@@ -27,18 +50,84 @@ var (
 		Description: "The number of HierConfig writes happened during HierConfig reconciliations",
 		Aggregation: ocview.LastValue(),
 	}
+
+	namespaceWritesView = &ocview.View{
+		Name:        "hnc/reconcilers/hierconfig/namespace_writes_total",
+		Measure:     namespaceWritesTotal,
+		Description: "The number of namespace writes happened during HierConfig reconciliations",
+		Aggregation: ocview.LastValue(),
+	}
+
+	objectReconcileTotalView = &ocview.View{
+		Name:        "hnc/reconcilers/object/total",
+		Measure:     objectReconcileTotal,
+		Description: "The total number of object reconciliations happened",
+		Aggregation: ocview.LastValue(),
+		TagKeys:     []tag.Key{KeyGK},
+	}
+
+	objectReconcileConcurrentView = &ocview.View{
+		Name:        "hnc/reconcilers/object/concurrent_peak",
+		Measure:     objectReconcileConcurrent,
+		Description: "The peak concurrent object reconciliations happened in the last reporting period",
+		Aggregation: ocview.LastValue(),
+		TagKeys:     []tag.Key{KeyGK},
+	}
+
+	objectWritesView = &ocview.View{
+		Name:        "hnc/reconcilers/object/object_writes_total",
+		Measure:     objectWritesTotal,
+		Description: "The number of object writes happened during object reconciliations",
+		Aggregation: ocview.LastValue(),
+		TagKeys:     []tag.Key{KeyGK},
+	}
 )
 
 func startRecordingMetrics() {
 	log := ctrl.Log.WithName("metricsRecorder")
 
-	// Register the view. It is imperative that this step exists,
+	// Register the views. It is imperative that this step exists,
 	// otherwise recorded metrics will be dropped and never exported.
-	if err := ocview.Register(hierReconcileView, hierWritesView); err != nil {
+	if err := ocview.Register(
+		hierReconcileTotalView,
+		hierReconcileConcurrentView,
+		hierWritesView,
+		namespaceWritesView,
+		objectReconcileTotalView,
+		objectReconcileConcurrentView,
+		objectWritesView,
+	); err != nil {
 		log.Error(err, "Failed to register the views")
 	}
+
+	go resetPeriodicMax()
+}
+
+func recordMetrics(m int64, ms *ocstats.Int64Measure) {
```

Comment:
Why is this `recordMetrics`, plural? Why not `recordMetric`?

Some comments about what each of these functions are for might be nice.